### ISSUE TITLE
pretrain accuracy

### ISSUE THREAD

**Issue by ruixuan-gao** [Tsingularity/FRN#8]
Dear author, sorry to bother you~ 
when I use pretrain model (/trained_model_weights/ResNet-12_finetune/model.pth) to train, I found that the accuracy is only 20%. I don't know what's wrong? Maybe I use false pretain model? I don't know~~
Maybe you have some good advice~
hope you have a good day~

**Comment by Tsingularity** [comment]
thanks for ur interest in our work!

could u please copy-paste the output training log here? thanks!

**Comment by Tsingularity** [comment]
and when u say "the accuracy", u mean the training accuracy in the log or the validation accuracy?

**Comment by ruixuan-gao** [comment]
thank you for your reply~
Although I change the format of log and some code which have no effect on the result, there are validation accuracy
![image](https://user-images.githubusercontent.com/55617926/168209907-17bff5ed-1ca8-4b50-9e67-ab2611d6210f.png)
thanks~

**Comment by Tsingularity** [comment]
could u please copy-paste more of the log? including both the training loss/acc and valid acc? thanks!

**Comment by ruixuan-gao** [comment]
ok~
when training epoch is 49 and 99.
![image](https://user-images.githubusercontent.com/55617926/168212835-e7b7e118-6656-4b7a-9921-544311e58920.png)
the loss function i used is cross - entropy rather than nllloss.
Maybe it will be clearer~
![image](https://user-images.githubusercontent.com/55617926/168213126-2daf15c3-0a72-4a38-ab9e-44894d6d9d48.png)
this is  variation trend.
sorry to bother you again^^

**Comment by Tsingularity** [comment]
from the log and figure u show above, looks like this is the training accuracy instead of the validation accuracy?

the training acc low might be reasonable because we are using high way during training (but i am not sure 20% is what we should expect, I'll check the fine-tuning log tmr on my end)

another thing i think might lead to this issue is replacing NLLLoss directly with CrossEntropyLoss. The model output is produced by LogSoftmax already and in this case, u are supposed to use NLLLoss. Check more in the pytorch [doc](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)

so I'll check this on my end tmr (fine-tune using the uploaded pre-trained model) and see how it goes. Meanwhile, could u please just run the original code and script as we provided on ur end and see whether it could produce good validation accuracy? Thanks!

**Comment by ruixuan-gao** [comment]
thanks for your reply~
I'll also check out some of my code. 
Many thanks for your kind and warm help ^^

**Comment by Tsingularity** [comment]
I just checked the history finetuning log on my end, for 5-shot 20-way fine-tuning, the training accuracy starts with around 80 and ends with around 90, and validation accuracy starts with around 82 and ends with around 85. So 20% accuracy is definitely something not normal.

I also noticed that you are actually using the final fine-tuned model (trained_model_weights/ResNet-12_finetune/model.pth) to do the training instead of the pre-trained weights we provide (trained_model_weights/mini-ImageNet/FRN/ResNet-12_pretrain/model.pth), but this shouldn't cause the low accuracy issue though.

In order to debug, I highly suggest u first run our original code by following the instructions in the readme, and see whether this could produce normal results, if yes, that probably means there're some issues with your modification and we could debug them together.

Feel free to let me know when u finish the running or if there's anything else I can help with.

**Comment by ruixuan-gao** [comment]
thank you sooooo much!
I am going to check my code and run code which you provided. And If I make any progress, I will tell you first time. 
Thank you again! And I'll keep getting better at this!!
Hope you have a good day! ^^

**Comment by Tsingularity** [comment]
no problem!

closing this issue for now, but feel free to re-open it if u have any more questions or updates on this issue.

**Comment by ruixuan-gao** [comment]
Hi~
I'm writing to tahnk you for your help and i'm very happy that my accuracy is finally getting close to yours!The pretrain - train accuracy are almost 90%, and val is 70%. It's little difference from yours because i think backbone i used is resnet18.&nbsp;
And the reason why 20% accuracy happened is dataset :( , i am so careless and&nbsp;entrenched ideas made me to set up error target.
Sorry to waste your time. thank you so much for your help and wish you every success :)




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年5月15日(星期天) 凌晨4:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Tsingularity/FRN] pretrain accuracy (Issue #8)





 
Closed #8.

**Comment by Tsingularity** [comment]
glad to hear that this issue is sloved and thanks for using our FRN!

**Comment by zhouyizhuo** [comment]
> Hi~ I'm writing to tahnk you for your help and i'm very happy that my accuracy is finally getting close to yours!The pretrain - train accuracy are almost 90%, and val is 70%. It's little difference from yours because i think backbone i used is resnet18.&nbsp; And the reason why 20% accuracy happened is dataset :( , i am so careless and&nbsp;entrenched ideas made me to set up error target. Sorry to waste your time. thank you so much for your help and wish you every success :)
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年5月15日(星期天) 凌晨4:19 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [Tsingularity/FRN] pretrain accuracy (Issue #8) Closed #8. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

@Glutaredoxin Hello, could you please share the specific reason why your training accuracy is normal but the verification accuracy is very low? I also encountered this problem when I was pre-training on my own dataset.😢The log file is as below：
![image](https://github.com/user-attachments/assets/0cd10357-0285-4582-ad42-7ceb0ed23d9e)
Any answer will be very helpful to me, thank you very much!😊o(*￣▽￣*)ブ

**Comment by Glutaredoxin** [comment]
Hi~
I think it was because of a small issue where I forgot to shuffle my dataset during pre-training. Maybe you could check if there's a problem with your dataset. Hope this helps, and good luck with your experiments :)
